internal/matcher/match.go

```diff
@@ -67,3 +71,173 @@ func Match(ctx context.Context, ir *claircore.IndexReport, matchers []driver.Mat
 	}
 	return vr, nil
 }
+
+// Store is the interface that can retrieve Enrichments and Vulnerabilities.
+type Store interface {
+	vulnstore.Vulnerability
+	vulnstore.Enrichment
+}
+
+// EnrichedMatch receives an IndexReport and creates a VulnerabilityReport containing matched vulnerabilities
+func EnrichedMatch(ctx context.Context, ir *claircore.IndexReport, ms []driver.Matcher, es []driver.Enricher, s Store) (*claircore.VulnerabilityReport, error) {
+	// the vulnerability report we are creating
+	vr := &claircore.VulnerabilityReport{
+		Hash:                   ir.Hash,
+		Packages:               ir.Packages,
+		Environments:           ir.Environments,
+		Distributions:          ir.Distributions,
+		Repositories:           ir.Repositories,
+		Vulnerabilities:        map[string]*claircore.Vulnerability{},
+		PackageVulnerabilities: map[string][]string{},
+		// The Enrichments member isn't constructed here because it's
+		// constructed separately and then added.
+	}
+	// extract IndexRecords from the IndexReport
+	records := ir.IndexRecords()
+	lim := runtime.GOMAXPROCS(0)
+
+	// Set up a pool to run matchers
```

Comment:
I think the admission of an error channel here probably makes sense given in that in the current code we're pushing to and just returning an error if one is pulled.